service/worker/scanner/scanner_test.go

```diff
@@ -0,0 +1,208 @@
+		TaskQueueName: executionsScannerTaskQueueName,
+	}
+	_ = executionScanner
+	taskQueueScanner := expectedScanner{
+		WFTypeName:    tqScannerWFTypeName,
+		TaskQueueName: tqScannerTaskQueueName,
+	}
+	historyScanner := expectedScanner{
+		WFTypeName:    historyScannerWFTypeName,
+		TaskQueueName: historyScannerTaskQueueName,
+	}
+
+	type testCase struct {
+		Name                     string
+		ExecutionsScannerEnabled bool
+		TaskQueueScannerEnabled  bool
+		HistoryScannerEnabled    bool
+		DefaultStore             string
+		ExpectedScanners         []expectedScanner
+	}
+
+	for _, c := range []testCase{
+		{
+			Name:                     "NothingEnabledNoSQL",
+			ExecutionsScannerEnabled: false,
+			TaskQueueScannerEnabled:  false,
+			HistoryScannerEnabled:    false,
+			DefaultStore:             config.StoreTypeNoSQL,
+			ExpectedScanners:         []expectedScanner{},
+		},
+		{
+			Name:                     "NothingEnabledSQL",
+			ExecutionsScannerEnabled: false,
+			TaskQueueScannerEnabled:  false,
+			HistoryScannerEnabled:    false,
+			DefaultStore:             config.StoreTypeSQL,
+			ExpectedScanners:         []expectedScanner{},
+		},
+		{
+			Name:                     "HistoryScannerNoSQL",
+			ExecutionsScannerEnabled: false,
+			TaskQueueScannerEnabled:  false,
+			HistoryScannerEnabled:    true,
+			DefaultStore:             config.StoreTypeNoSQL,
+			ExpectedScanners:         []expectedScanner{historyScanner},
+		},
+		{
+			Name:                     "HistoryScannerSQL",
+			ExecutionsScannerEnabled: false,
+			TaskQueueScannerEnabled:  false,
+			HistoryScannerEnabled:    true,
+			DefaultStore:             config.StoreTypeSQL,
+			ExpectedScanners:         []expectedScanner{historyScanner},
+		},
+		{
+			Name:                     "TaskQueueScannerNoSQL",
+			ExecutionsScannerEnabled: false,
+			TaskQueueScannerEnabled:  true,
+			HistoryScannerEnabled:    false,
+			DefaultStore:             config.StoreTypeNoSQL,
+			ExpectedScanners:         []expectedScanner{}, // TODO: enable task queue scanner for NoSQL?
+		},
+		{
+			Name:                     "TaskQueueScannerSQL",
+			ExecutionsScannerEnabled: false,
+			TaskQueueScannerEnabled:  true,
+			HistoryScannerEnabled:    false,
+			DefaultStore:             config.StoreTypeSQL,
+			ExpectedScanners:         []expectedScanner{taskQueueScanner},
+		},
+		{
+			Name:                     "ExecutionsScannerNoSQL",
+			ExecutionsScannerEnabled: true,
+			TaskQueueScannerEnabled:  false,
+			HistoryScannerEnabled:    false,
+			DefaultStore:             config.StoreTypeNoSQL,
+			ExpectedScanners:         []expectedScanner{executionScanner},
+		},
+		{
+			Name:                     "ExecutionsScannerSQL",
+			ExecutionsScannerEnabled: true,
+			TaskQueueScannerEnabled:  false,
+			HistoryScannerEnabled:    false,
+			DefaultStore:             config.StoreTypeSQL,
+			ExpectedScanners:         []expectedScanner{executionScanner},
+		},
+	} {
```

Comment:
Plz add a test case where store type is SQL and all three scanners are enabled.

Reply:
Will do